assets/js/modules/analytics-4/datastore/webdatastreams.test.js

```diff
@@ -159,5 +159,44 @@ describe( 'modules/analytics-4 webdatastreams', () => {
 				expect( console ).toHaveErrored();
 			} );
 		} );
+
+		describe( 'getMatchingWebDataStream', () => {
+			const propertyID = '12345';
+
+			it( 'should return undefined if web data streams arent loaded yet', () => {
+				freezeFetch( webDataStreamsEndpoint );
+
+				const datastream = registry.select( STORE_NAME ).getMatchingWebDataStream( propertyID );
+				expect( datastream ).toBeUndefined();
+			} );
+
+			it( 'should return NULL when no datastreams are matched', () => {
+				provideSiteInfo( registry, { referenceSiteURL: 'http://example.net' } );
+				registry.dispatch( STORE_NAME ).receiveGetWebDataStreams( fixtures.webDataStreams, { propertyID } );
+
+				const datastream = registry.select( STORE_NAME ).getMatchingWebDataStream( propertyID );
+				expect( datastream ).toBeNull();
+			} );
+
+			it( 'should return the correct datastream when reference site URL matches exactly', () => {
+				provideSiteInfo( registry, { referenceSiteURL: 'http://example.com' } );
+				registry.dispatch( STORE_NAME ).receiveGetWebDataStreams( fixtures.webDataStreams, { propertyID } );
+
+				const datastream = registry.select( STORE_NAME ).getMatchingWebDataStream( propertyID );
+				expect( datastream ).toEqual( fixtures.webDataStreams[ 0 ] );
```

Comment:
I would prefer not to reference individual items from fixtures like this (same for the `it.each` below). These fixtures are mostly for use in tests which need a fetch mock with a decent sized object. It would be a bit more useful for the test to be able to see the definition of the webdatastream object - particularly since we're not using a factory to create it and the specific object you're referencing is significant. This was why builders were introduced for GTM (which was done after Analytics – it would be great to eventually do the same there though); not sure if it makes sense for GA4 or not; what do you think?

The simplest thing to do here would be to add definitions for these in the describe above (or at the top) for each of the two webdatastream objects and then receive them more explicitly in the test:

E.g.
```js
// above
const webDataStreamDotCom = { ... };
const webDataStreamDotOrg = { ... };
const webDataStreams = [ webDataStreamDotCom, webDataStreamDotOrg ];
// in test:
registry.dispatch( STORE_NAME ).receiveGetWebDataStreams( webDataStreams, { propertyID } );
// ...
expect( datastream ).toEqual( webDataStreamDotOrg );
```

This would be easier to reason about.

Reply:
Updated.